airbyte-integrations/connectors/source-mongodb-v3/src/main/kotlin/io/airbyte/integrations/source/mongodb/MongoDbFieldTypes.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.mongodb
+
+import io.airbyte.cdk.data.AirbyteSchemaType
+import io.airbyte.cdk.data.BinaryCodec
+import io.airbyte.cdk.data.BooleanCodec
+import io.airbyte.cdk.data.JsonDecoder
+import io.airbyte.cdk.data.JsonEncoder
+import io.airbyte.cdk.data.JsonStringCodec
+import io.airbyte.cdk.data.LeafAirbyteSchemaType
+import io.airbyte.cdk.data.NullCodec
+import io.airbyte.cdk.discover.LosslessFieldType
+import java.nio.ByteBuffer
+
+data object MongoBooleanFieldType : LosslessFieldType {
```

Comment:
you might want to have these all inherit from some `sealed interface MongoFieldType<T>` and the type parameter might make this easier too.